File: frontend/src/types/analytics.ts
// Copyright (c) 2026 Mechemsi. All rights reserved.
// Licensed under AGPLv3. See LICENSE file.
// Commercial licensing: info@mechemsi.com

export interface PhaseDurationStat {
  phase_name: string;
  avg_seconds: number;
  count: number;
}

export interface AgentStat {
  agent_name: string;
  total_runs: number;
  success_rate: number;
  avg_duration_seconds: number;
}

export interface DailyRunCount {
  date: string;
  count: number;
}

export interface AgentCostStat {
  agent_name: string;
  cost_usd: number;
}

export interface CostStats {
  total_cost_usd: number;
  total_tokens_in: number;
  total_tokens_out: number;
  avg_cost_per_run_usd: number;
  cost_by_agent: AgentCostStat[];
}

export interface AnalyticsSummary {
  success_rate: number;
  avg_duration_seconds: number;
  total_runs: number;
  runs_by_status: Record<string, number>;
  avg_phase_durations: PhaseDurationStat[];
  agent_stats: AgentStat[];
  runs_over_time: DailyRunCount[];
  cost_stats?: CostStats;
}

export interface Stats {
  total_runs: number;
  pending: number;
  running: number;
  awaiting_approval: number;
  completed: number;
  failed: number;
}

File: backend/schemas/analytics.py
# Copyright (c) 2026 Mechemsi. All rights reserved.
# Licensed under AGPLv3. See LICENSE file.
# Commercial licensing: info@mechemsi.com

from pydantic import BaseModel


class PhaseDurationStat(BaseModel):
    phase_name: str
    avg_seconds: float
    count: int


class AgentStat(BaseModel):
    agent_name: str
    total_runs: int
    success_rate: float
    avg_duration_seconds: float


class DailyRunCount(BaseModel):
    date: str
    count: int


class AgentCostStat(BaseModel):
    agent_name: str
    cost_usd: float


class CostStats(BaseModel):
    total_cost_usd: float
    total_tokens_in: int
    total_tokens_out: int
    avg_cost_per_run_usd: float
    cost_by_agent: list[AgentCostStat]


class AnalyticsSummary(BaseModel):
    success_rate: float
    avg_duration_seconds: float
    total_runs: int
    runs_by_status: dict[str, int]
    avg_phase_durations: list[PhaseDurationStat]
    agent_stats: list[AgentStat]
    runs_over_time: list[DailyRunCount]
    cost_stats: CostStats | None = None


class StatsResponse(BaseModel):
    total_runs: int
    pending: int
    running: int
    awaiting_approval: int
    completed: int
    failed: int
